examples/Substrait.ipynb

```diff
@@ -0,0 +1,254 @@
+    "\n",
+    "### Why Substrait?\n",
+    "\n",
+    "The current state of the world requires tools like Ibis to build connectors for each unique data system. This is a many-to-many relationship that grows exponentiatlly. Substrait removes the need for connectors by introducing an Intermediate Representation (IR). Now, we can have a many-to-one relationship from frontend -> IR and a one-to-many relationship from IR -> backend. \n",
+    "\n",
+    "### But, how is this useful to me?\n",
+    "\n",
+    "Interoperability now _and in the future_. The same Substrait Plan can run anywhere that has built-in support for the Substrait specification. No need to wait for Ibis to implement the shiny new connector for your data system of choice."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Example\n",
+    "\n",
+    "Let's see Ibis Substrait in action."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Setup\n",
+    "\n",
+    "Let's build a toy example of a database server. Our example uses a local DuckDB database, but in practice we can imagine talking to a database server over the network."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import duckdb\n",
+    "import os\n",
+    "from urllib.request import urlretrieve\n",
+    "\n",
+    "\n",
+    "class DatabaseServer:\n",
+    "    DB_NAME = \"palmer_penguins.ddb\"\n",
+    "    DB_URL = \"https://storage.googleapis.com/ibis-tutorial-data/palmer_penguins.ddb\"\n",
+    "    \n",
+    "    def __init__(self):\n",
+    "        if not os.path.exists(self.DB_NAME):\n",
+    "            urlretrieve(self.DB_URL, self.DB_NAME)\n",
+    "        self.db = duckdb.connect(self.DB_NAME)\n",
+    "        self.db.install_extension(\"substrait\")\n",
+    "        self.db.load_extension(\"substrait\")\n",
+    "    \n",
+    "    def execute(self, substrait):\n",
+    "        result = self.db.from_substrait(substrait)\n",
+    "        return result.fetchall()\n",
+    "    \n",
+    "    @property\n",
+    "    def connection_string(self):\n",
+    "        \"\"\"Helper for our toy example\"\"\"\n",
+    "        return self.DB_NAME\n",
+    "        \n",
+    "\n",
+    "db_server = DatabaseServer()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Ibis Table\n",
+    "\n",
+    "We need an Ibis Table to query against. We'll take a shortcut in our example by querying our local DB directly. \n",
+    "\n",
+    "Since we only need the table name and schema, we acquire the table through our backend connection and then call [unbind()](https://ibis-project.org/api/expressions/?h=unbind#ibis.expr.types.core.Expr.unbind). This will remove any backend-specific objects."
```

Comment:
I think it might be better to define the penguins schema manually, so that we don't conflate connecting "directly" to the DuckDB data vs. via substrait.

Reply:
Will do! I agree